src/librustc/middle/resolve_lifetime.rs

```diff
@@ -748,7 +747,7 @@ fn object_lifetime_defaults_for_item(hir_map: &Map, generics: &hir::Generics)
         match set {
             Set1::Empty => Set1::Empty,
             Set1::One(name) => {
-                if name == keywords::StaticLifetime.name() {
+                if name == "'static" {
```

Comment:
This seems unfortunate - can we at least make this a single const rather than a string literal?